openedx/core/djangoapps/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -1759,26 +1770,43 @@ def test_get_tags(self):
         assert status.is_success(response3.status_code)
         assert response3.data[str(self.courseA)]["taxonomies"] == expected_tags
 
-    def test_object_tags_query_count(self):
+    @ddt.data(
+        ('staff', 'courseA', 7),
+        ('staff', 'libraryA', 7),
+        ("content_creatorA", 'courseA', 13, False),
+        ("content_creatorA", 'libraryA', 13, False),
+        ("library_staffA", 'libraryA', 13, False),  # Library users can only view objecttags, not change them?
+        ("library_userA", 'libraryA', 13, False),
+        ("instructorA", 'courseA', 13),
+        ("course_instructorA", 'courseA', 13),
+        ("course_staffA", 'courseA', 13),
+    )
+    @ddt.unpack
+    def test_object_tags_query_count(
+            self,
+            user_attr: str,
+            object_attr: str,
+            expected_queries: int,
+            expected_perm: bool = True):
         """
         Test how many queries are used when retrieving object tags and permissions
         """
-        object_key = self.courseA
+        object_key = getattr(self, object_attr)
         object_id = str(object_key)
         tagging_api.tag_object(object_id=object_id, taxonomy=self.t1, tags=["anvil", "android"])
         expected_tags = [
-            {"value": "android", "lineage": ["ALPHABET", "android"], "can_delete_objecttag": True},
-            {"value": "anvil", "lineage": ["ALPHABET", "anvil"], "can_delete_objecttag": True},
+            {"value": "android", "lineage": ["ALPHABET", "android"], "can_delete_objecttag": expected_perm},
+            {"value": "anvil", "lineage": ["ALPHABET", "anvil"], "can_delete_objecttag": expected_perm},
         ]
-
         url = OBJECT_TAGS_URL.format(object_id=object_id)
-        self.client.force_authenticate(user=self.staff)
-        with self.assertNumQueries(7):  # TODO Why so many queries?
+        user = getattr(self, user_attr)
+        self.client.force_authenticate(user=user)
+        with self.assertNumQueries(expected_queries):
```

Comment:
Same question here - doesn't seem like the # of queries changed.

Reply:
If you expand the notes in the PR description, you can see details about what queries are still being executed.